tls/ttls.c

```diff
@@ -1049,11 +1056,6 @@ ttls_hdr_check(TlsCtx *tls)
 		T_DBG("minor version mismatch %d\n", tls->minor);
 		return T_DROP;
 	}
-	/* Check length against the size of our buffer */
-	if (unlikely(io->msglen > TTLS_PAYLOAD_LEN)) {
-		T_DBG("too big message length: %u\n", io->msglen);
-		return T_DROP;
-	}
```

Comment:
We first read `msglen` in `ttls_parse_record_hdr()` and we must check the record length as early as possible and it's now enough to check the length in `ttls_decrypt()`, e.g. it seems now we go to handshake FSM w/o the length checking.

Reply:
If you open the file context below, you'll see that there are no calls to `ttls_decrypt()` there. Instead, `io->msgtype` is almost immediately checked again. But with the knowledge of current state, with encryption or not, the checks are more precise. I think this particular check is redundant, as it doesn't prevent any costly calculations anyway.